pkg/wellknownoperators/operators.go

```diff
@@ -38,65 +40,183 @@ type Builder struct {
 	Cluster *kops.Cluster
 }
 
-func (b *Builder) Build() ([]*WellKnownAddon, kubemanifest.ObjectList, error) {
+func (b *Builder) Build(objects kubemanifest.ObjectList) ([]*Package, kubemanifest.ObjectList, error) {
 	if !featureflag.UseAddonOperators.Enabled() {
-		return nil, nil, nil
+		return nil, objects, nil
 	}
 
-	var addons []*WellKnownAddon
-	var crds kubemanifest.ObjectList
+	var packages []*Package
+	var keepObjects kubemanifest.ObjectList
+
+	for _, object := range objects {
+		keep := true
+
+		// We may in future move ClusterPackage to the server-side,
+		// however remapping it client-side allow us to use our existing manifest logic
```

Comment:
Does this support rendering manifest templates using values from the cluster spec? (or a different API type?) how do we plan to offer the equivalent level of customizability that we have now with addons and their api types?

Reply:
So generally my hope/expectation is that we don't customize the operator itself; by design it's supposed to be very generic.

The CRD-instance _would_ offer the customization.  Because that CRD is an operator-specific CRD (e.g. CoreDNS or KopeioNetworking) it has a specific schema for that customization.  Eventually I hope we get to a world where the user configures those addon CRD instances without kops even being aware of what's going on.

Where today we have configuration in the Cluster, we have to map that to a CRD.  Eventually we can have a migration tool or similar to get the values out of the Cluster, but today we can have the values specified in the Cluster generate the CRDs.  We don't have that yet / in this PR.

Where the CRD / operator has dependencies on other objects, the hope is that it can get those from the cluster itself.  For example, today the CoreDNS operator needs the DNS service IP address, and so the operator can look up the kube-dns service to default the value.